adr/ADR-44.md

```diff
@@ -0,0 +1,109 @@
+and no messages should be delivered to any consumers, messages can't be deleted, configuration cannot be updated - 
+it is offline in every way that would result in a change.  Likewise a compatible offline mode should exist for Consumers. 
+
+The Stream and Consumer state should get new fields `Offline bool` and `OfflineReason string` that should be set for 
+such assets.
+
+When the server starts and determines it cannot start an asset for any reason it should set this mode and fields.
+
+For starting incompatible streams in offline mode we would need to load the config in the current manner to figure out
+which subjects Streams would listen on since even while Streams are offline we do need the protections of
+overlapping subjects to be active to avoid issues later when the Stream can come online again.
+
+# Safe unmarshalling of JSON data
+
+The JetStream API and Meta-layer should start using the `DisallowUnknownFields` feature in the go json package and 
+detect when asked to load incompatible assets or serve incompatible API calls and should error in the case of the 
+API and start assets in Offline mode in the case of assets.
+
+This will prevent assets inadvertently reverting some settings and changing behaviour during downgrades.
+
+# Minimal supported server version for assets
+
+When the server loads assets it should detect incompatible features using a combination of `DisallowUnknownFields` 
+and comparing the server versions that created assets on a major.minor level only.
+
+Incompatible assets should be loaded in Offline mode.
+
+# Concerns
+
+The problem with this approach is that we only know if something worked, or was compatible with the server, after the
+asset was created. In cases where a feature adds new configuration fields this would be easily detected by the 
+Marshaling but on more subtle features like previous changes to `DiscardNew` the client would need to verify the 
+response to ensure the version requirements were met and then remove the bad asset, this is a pretty bad error 
+handling scenario.
+
+This can be slightly mitigated by including the current server version in the `$JS.API.INFO` response so at least 
+the meta leader version is known - but in reality one cannot really tell a lot from this version since it is not 
+guaranteed to be representative of the cluster and is particularly problematic in long running clients as the server
+may have been upgraded since last `INFO` call.
+
+An alternative would be that the server maintains an API compatability counter that increments when we add features.  
+To use a feature the clients would send something that indicates the minimum required feature level and the server 
+would fail to add it. While this would solve the problem it would be hugely problematic to make work in all the 
+clients without bugs.
+
+# Implementation Targets
+
+For 2.11 we should start reporting the metadata and possibly support offline assets, we might only do the remaining
```

Comment:
Implementation wise, `DisallowUnknownFields` sounds like the easiset thing to implement, but at the same time, it has the biggest risk of breaking clients.

Reply:
indeed its the biggest risk here